logstash-core/src/main/java/org/logstash/config/ir/compiler/AbstractFilterDelegatorExt.java

```diff
@@ -39,7 +39,7 @@ public AbstractFilterDelegatorExt(final Ruby runtime, final RubyClass metaClass)
         super(runtime, metaClass);
     }
 
-    protected void initMetrics(final String id, final AbstractNamespacedMetricExt namespacedMetric) {
+    protected void initMetrics(final String id, final AbstractNamespacedMetricExt namespacedMetric, String codeRef) {
```

Comment:
hmm, seems like you will always convert this arg to a RubyString why not pass down a RubyString ?

Reply:
No in `JavaOutputDeletegatorExt.create` receive Java string created https://github.com/elastic/logstash/pull/11288/files/c347a1dd45728d08618ebf40f3b62f6fbfa15124#diff-f99a4d603d2230e54ef9d84b92f5d36dR239-R244

Reply:
yes but this PR is in charge of that change I meant: why not change `String configRef` -> `RubyString configRef`
and than `initMetrics` could also receive a `RubyString` what we're doing here on a few places is : 
get a `RubyString` convert to a `String` (using `asJavaString()`) and than back `RubyString.newString(...)`

Reply:
(might be missing some parts here - please let me know if its not feasible to do so)